src/Exp.js

```diff
@@ -602,6 +603,18 @@ class Exp {
         });
     }
 
+    /* Method for updating exp-ifs methods */
+    updateIfsMethods() {
+        const expIfs = this.select(`*[exp-if]`);
+        expIfs.forEach(el => {
+            const key = el.getAttribute('exp-if');
+            if (key in this.methods) {
```

Comment:
`methods can be replaced and modified dynamically` this should not be done. See The Exp instance in index.html documentation.

`When this data changes, the view will re-render. It should be noted that properties in data are only reactive if they existed when the instance was created. `

This should be the same for `methods` too not only `data`.

Also this.methods will be updated when you redefine a method after Exp has been initialized. Methods in model are a reference to this.methods object, not a copy.

Actually I don't see any added value in using `this.model[key]`.

`(key in this.methods)` makes more sense semantically too.  _I am looking for any key that was defined as a method._ and not _I am looking for any key that has type of function_

Reply:
*modifying and replacing methods* I agree that it should not be done, but that it how Exp is designed right now, so lets obey it, and change it to static methods in a separate issue. Right now always methods are called from the model, and even my PR with the exp-if calling methods is calling them from model.

*Also this.methods will be updated when you redefine a method after Exp has been initialized. Methods in model are a reference to this.methods object, not a copy.* Be careful that is not true. Methods are copied into model at line 300-304 of `initializeModel` function and try to run this script. 
```
var x = { foo: () => 1 }
var y = { foo: x["foo"] }
x["foo"] = () => 2;
console.log(x["foo"]())
console.log(y["foo"]())
```
JavaScript copies by value, it does not reference it.

Reply:
I see, my bad. I just don't like checking by `function`. I would rather check if the key was defined in method, which my code does. Even if Exp was designed in certain way, we should not encourage "bad practice" think about backwards compatibility. If someone dynamically redefines a method (which is not only wrong in terms of Exp but wrong in general) then we change it to static methods, the banner will break.

Reply:
Ok. so how about you change it to model right now, release v1.2.2 and then raise a new issue where we change all occurrences of this and have methods be called from `this.methods` rather than `this.model`. I agree with you, I just want to obey code integrity and not to have code behave in one place differently than in the other